core/src/test/java/org/elasticsearch/index/query/ConstantScoreQueryBuilderTests.java

```diff
@@ -67,15 +68,15 @@ public void testFilterElement() throws IOException {
     }
 
     /**
-     * test that multiple "filter" elements causes {@link ParsingException}
+     * test that multiple "filter" elements causes {@link JsonParseException}
      */
     public void testMultipleFilterElements() throws IOException {
         String queryString = "{ \"" + ConstantScoreQueryBuilder.NAME + "\" : {\n" +
                                     "\"filter\" : { \"term\": { \"foo\": \"a\" } },\n" +
                                     "\"filter\" : { \"term\": { \"foo\": \"x\" } },\n" +
                             "} }";
-        ParsingException e = expectThrows(ParsingException.class, () -> parseQuery(queryString));
-        assertThat(e.getMessage(), containsString("accepts only one 'filter' element"));
+        JsonParseException e = expectThrows(JsonParseException.class, () -> parseQuery(queryString));
```

Comment:
Same here.

I wonder if it makes more sense to leave these tests as they are and skip them if the value of the duplicate check property is `true`. That way if we really want we can set it to false and run the tests again. And if they have a compile time reference to the duplicate check property when we remove the duplicate check property the test will stop compiling and we can use that as a reminder to remove that particular manual duplicate check.